src/components/closed-captions/closed-captions.js

```diff
@@ -0,0 +1,71 @@
+//@flow
+import style from '../../styles/style.scss';
+import {h} from 'preact';
+import {useEffect, useState} from 'preact/hooks';
+import {withText} from 'preact-i18n';
+import {default as Icon, IconType} from '../icon';
+import {connect} from 'react-redux';
+import {withEventDispatcher} from 'components/event-dispatcher';
+import {withLogger} from 'components/logger';
+import {Tooltip} from 'components/tooltip';
+import {Button} from 'components/button';
+import {ButtonControl} from 'components/button-control';
+/**
+ * mapping state to props
+ * @param {*} state - redux store state
+ * @returns {Object} - mapped state to this component
+ */
+const mapStateToProps = state => ({
+  textTracks: state.engine.textTracks,
+  showCCButton: state.config.showCCButton
+});
+
+const COMPONENT_NAME = 'ClosedCaptions';
+
+/**
+ * ClosedCaptions component
+ *
+ * @class ClosedCaptions
+ * @example <ClosedCaptions />
+ * @extends {Component}
+ */
+const ClosedCaptions = connect(mapStateToProps)(
+  withLogger(COMPONENT_NAME)(
+    withEventDispatcher(COMPONENT_NAME)(
+      withText({
+        closedCaptionsOnText: 'controls.closedCaptionsOn',
+        closedCaptionsOffText: 'controls.closedCaptionsOff'
+      })((props, context) => {
+        const [ccOn, setCCOn] = useState(false);
+        const {textTracks} = props;
+        const {player} = context;
+        const activeTextTrack = textTracks.find(textTrack => textTrack.active);
+
+        useEffect(() => {
+          setCCOn(activeTextTrack?.language !== 'off');
+        }, [activeTextTrack]);
+
+        return props.textTracks?.length && props.showCCButton ? (
```

Comment:
since it's one component with different props for on/off, I would just extract it to a separate function and then do something like

if (!props.textTracks?.length && props.showCCButton) return undefined;

return ccOn ? <component {...props1}> : <component {props2}>

I like it better than having the same if in several places